ext/standard/streamsfuncs.c

```diff
@@ -756,115 +782,132 @@ static int stream_array_emulate_read_fd_set(zval *stream_array)
 /* }}} */
 
```

Comment:
Can this be tested from useland when more than 1024 file descriptors are created/used?

Reply:
Also when there is less than 1024, but at least one has a number > 1024

Reply:
@mvorisek I used this code to test:

```php
<?php
// Create a large number of file descriptors
$files = [];
$r = $w = $e = [];

for ($i = 0; $i < 20000; $i++) {
    $fileName = tempnam(sys_get_temp_dir(), 'testfile');
    $fp = fopen($fileName, 'r');
    if ($fp === false) {
        die("Failed to open file: $fileName\n");
    }
    $files[] = ['name' => $fileName, 'handle' => $fp];
    $r[] = $fp;
}

// Check the number of file descriptors
$res = stream_select($r, $w, $e, 0, 0);
var_dump($res, $r);
// Clean up
foreach ($files as $file) {
    fclose($file['handle']);
    unlink($file['name']);
}
```

Reply:
A function doing `dup2()` can probably be added to `zend_test`, allowing to `zend_test_dup2(1, 9999)`.

Reply:
@TimWolla Is zend_test a mechanism to add special functions to the PHP runtime during testing? I think it would be nice to have zend_test_get_fd($resource), but perhaps that already exists?

Reply:
@frodeborli 

> Is zend_test a mechanism to add special functions to the PHP runtime during testing?

Yes, exactly. It is meant to provide helpers to test stuff that would otherwise be untestable. You can find it in https://github.com/php/php-src/tree/master/ext/zend_test and can enable it at build time using `--enable-zend-test`.